### PR TITLE
chore(repo): maintainers can run sdk spec tests on forks

### PR DESCRIPTION
**Pull request by tsuf239**
fixes [#4139](https://github.com/winglang/wing/issues/4139)
https://www.loom.com/share/5b33e2635ed04be2a8502d7fc5f1116f
## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: .github/workflows/tf-aws-test.yml [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
   NODE_VERSION: "18.16.0"
   # this variable indicates wheater to use wing cli from a local version (using the current repo code)
   # or use the latest remote npm version
-  LOCAL_BUILD: ${{ github.workflow == 'build'}}
+  LOCAL_BUILD: ${{ github.workflow == 'build' && github.event_name == 'workflow_dispatch'}}
+  # indicates if the action triggered from a workflow_dispatch
+  MANUAL: ${{ github.event_name == 'workflow_dispatch' }}
+  REPO: ${{inputs.repo || github.repository}}
+  REF: ${{inputs.ref || github.ref}}
+  ALLOWED_USERS: ("tsuf239" "MarkMcCulloh" "eladb" "revitalbarletz" "ShaiBer" "Chriscbr" "hasanaburayyan" "staycoolcall911" "ekeren")
```
Comment (staycoolcall911) [inline .github/workflows/tf-aws-test.yml:24]: I hope there's a better way to specify the @mainters - perhaps someone would have a concrete idea.
In any case, this list is missing names

**Comment by eladb** [comment]
This should be a "chore" because it's not a new feature of winglang.

File: .github/workflows/tf-aws-test.yml [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+        type: string
+        default: main
 
 env:
   AWS_REGION: "us-east-1"
   NODE_VERSION: "18.16.0"
   # this variable indicates wheater to use wing cli from a local version (using the current repo code)
   # or use the latest remote npm version
-  LOCAL_BUILD: ${{ github.workflow == 'build'}}
+  LOCAL_BUILD: ${{ github.workflow == 'build' && github.event_name == 'workflow_dispatch'}}
```
[the remark this answers is not in the document]
Comment (tsuf239) [reply .github/workflows/tf-aws-test.yml:19]: Exactly!

File: .github/workflows/tf-aws-test.yml [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+  REPO: ${{inputs.repo || github.repository}}
+  REF: ${{inputs.ref || github.ref}}
+  ALLOWED_USERS: ("tsuf239" "MarkMcCulloh" "eladb" "revitalbarletz" "ShaiBer" "Chriscbr" "hasanaburayyan" "staycoolcall911" "ekeren")
+  PNPM_VERSION: "8.6.3"
 
 jobs:
   setup:
     runs-on: ubuntu-latest
     steps:
+      - name: cancel run if not allowed
```
[the remark this answers is not in the document]
Comment (tsuf239) [reply .github/workflows/tf-aws-test.yml:31]: Uri and I didn't manage to get the token :/ 
https://github.com/tspascoal/get-user-teams-membership#requirements
Can you help with that, maybe?

[the remark this answers is not in the document]
Comment (MarkMcCulloh) [reply .github/workflows/tf-aws-test.yml:31]: You can probably use `secrets.PROJEN_GITHUB_TOKEN` as long is the usage of it only happens during workflow_dispatch

File: .github/workflows/tf-aws-test.yml [hunk 4]
```diff
@@ -1,21 +1,44 @@
...
   workflow_call: {}
-  workflow_dispatch: {}
+  workflow_dispatch:
+    inputs:
+      repo:
+        type: string
+        default: winglang/wing
+      ref:
+        type: string
+        default: main
```
Comment (MarkMcCulloh) [inline .github/workflows/tf-aws-test.yml:12]:
```suggestion
      repo:
        type: string
        default: winglang/wing
        description: "Full repo name (owner/name)"
      ref:
        type: string
        default: main
        description: "Ref to checkout. Must be present in the repo input."
```

**Review by MarkMcCulloh — APPROVED** [review]
Other than removing the hardcoded usernames, LGTM

File: .github/workflows/tf-aws-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsuf239) [reply .github/workflows/tf-aws-test.yml:40]: Added a new one :) 
it's working now: 
image
removing the list of users then

**Pull request merged**